src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -260,10 +270,41 @@ public bool HasUnsavedChanges
 
         /// <summary>
         ///     All of the nodes currently in the workspace.
-        /// 
-        ///     TODO(Peter): This should be an IEnumerable of nodes to prevent modification from the outside - MAGN-6580
         /// </summary>
-        public ObservableCollection<NodeModel> Nodes { get { return nodes; } }
+        public IEnumerable<NodeModel> Nodes { 
+            get 
+            {
+                IEnumerable<NodeModel> nodesClone;
+                lock (nodes)
+                {
+                    nodesClone = nodes.ToList();                
+                }
+
+                return nodesClone;
+            } 
+        }
+
+
```

Comment:
They're placed in local proximity to where they were. I will move them, but that is equally inconsistent under another set of parameters.